devel/MacOSX.sdk/Portfile

```diff
@@ -4,7 +4,7 @@ PortSystem              1.0
 
 name                    MacOSX.sdk
 categories              devel
-platforms               macosx
+platforms               {macosx any}
```

Comment:
This is a good idea.

It will trigger rebuilds of every subport so I will need to load the distfiles that can't be downloaded automatically onto one of the build machines manually before merging this PR.

Reply:
Unfortunately it looks like this port has bitrotted. For newer SDKs it downloaded them direct from Apple's CDN but now Apple is delivering a 403 response, e.g.:

```
% curl -I https://swcdn.apple.com/content/downloads/28/19/071-82714-A_HUFJRGID6Y/nnmlg28zs93jdbrsqtk6krr4mmaize40hb/CLTools_macOSLMOS_SDK.pkg

HTTP/1.1 403 Forbidden
…
```

so we either need a new URL from which to download the packages that used to be available via Apple's CDN or we need to revert to the old strategy of manually downloading the CLT dmgs from the Apple Developer web site and adjust the port's programming accordingly.

Reply:
How does one determine the proper URL from swcdn.apple.com?
There are various posts with URLs from swcdn.apple.com, but none of them seem to mention how to find them.

Reply:
I probably had just found one of those posts at the time. I don't remember.

Reply:
It would seem that someone has taken the time to [search for this information](https://github.com/zhangyoufu/swscan.apple.com/blob/master/url.txt).
I have updated the pull request to remove nonexistent files.
However, I believe the fix (https://github.com/macports/macports-ports/pull/23312/commits/e48c3364486be4cd756ad4a3b7605b3d12d3c8ac) constitutes a stealth update, but fixing it would disrupt the existing distfile naming convention. 

I also added SDKs for macOS 13-14 (https://github.com/macports/macports-ports/pull/23312/commits/fe83f2fd0e7bd4b04d731c596c2409129e83d3dd), but I do not know how to find the release dates of the distfiles.

Reply:
> I believe the fix ([e48c336](https://github.com/macports/macports-ports/commit/e48c3364486be4cd756ad4a3b7605b3d12d3c8ac)) constitutes a stealth update, but fixing it would disrupt the existing distfile naming convention.

And the SDK version is still exactly the same? Can you compare the old and new distfiles to see what the difference is? If you don't have the old distfiles, I can check if I saved them, which I was in the habit of doing for this port.

This will fix https://trac.macports.org/ticket/69952 by the way.

> I also added SDKs for macOS 13-14 ([fe83f2f](https://github.com/macports/macports-ports/commit/fe83f2fd0e7bd4b04d731c596c2409129e83d3dd)), but I do not know how to find the release dates of the distfiles.

You could check if there is an HTTP Last-Modified header. If not, you could check the Internet; if not Apple, then surely somebody has posted something to announce these releases. You could also look at the modification date of the dmg or the files in it.

Reply:
> And the SDK version is still exactly the same? Can you compare the old and new distfiles to see what the difference is? If you don't have the old distfiles, I can check if I saved them, which I was in the habit of doing for this port.

As far as I can tell, they have the same SDK version.
Unfortunately, I do not have access to the old distfiles, so I do not know what the differences are.
 
> This will fix https://trac.macports.org/ticket/69952 by the way.

Thank you for pointing this out.
I have updated the commit message.

> You could check if there is an HTTP Last-Modified header. If not, you could check the Internet; if not Apple, then surely somebody has posted something to announce these releases. You could also look at the modification date of the dmg or the files in it.

I have updated the dates using the HTTP Last-Modified header.